src/mrack/providers/beaker.py

```diff
@@ -0,0 +1,334 @@
+    "Cancelled": STATUS_DELETED,
+    "Aborted": STATUS_ERROR,
+    "Completed": STATUS_OTHER,
+}
+
+
+class BeakerProvider(Provider):
+    """Beaker Provider."""
+
+    def __init__(self):
+        """Object initialization."""
+        self._name = PROVISIONER_KEY
+        self.conf = PyConfigParser()
+        self.timeout = 60  # minutes
+        self.poll_sleep = 30  # seconds
+        self.max_attempts = 240  # 240 * 30s timeout - 2h timeout
+
+    async def init(self, distros):
+        """Initialize provider with data from Beaker configuration."""
+        logger.info("Initializing Beaker provider")
+        login_start = datetime.now()
+        default_config = os.path.expanduser(
+            os.environ.get("BEAKER_CONF", "/etc/beaker/client.conf")  # TODO use provc
+        )
+        self.conf.load_from_file(default_config)
+        self.hub = HubProxy(logger=logger, conf=self.conf)
+        login_end = datetime.now()
+        login_duration = login_end - login_start
+        logger.info(f"Login duration {login_duration}")
+
+    async def validate_hosts(self, hosts):
+        """Validate that host requirements are well specified."""
+        return
+
+    def _allow_ssh_key(self, ssh_key):
+
+        with open(ssh_key, "r") as key_file:
+            key_content = key_file.read()
+
+        return [
+            """%%post
+mkdir -p /root/.ssh
+cat >>/root/.ssh/authorized_keys << "__EOF__"
+%s__EOF__
+restorecon -R /root/.ssh
+chmod go-w /root /root/.ssh /root/.ssh/authorized_keys
+%%end"""
+            % "".join(key_content)
+        ]
+
+    def _req_to_bkr_job(self, req):
+        specs = deepcopy(req)  # work with own copy, do not modify the input
+
+        # Job attributes:
+        specs.update({"retention_tag": "audit"})
+        specs.update({"product": "[internal]"})
+        specs.update({"whiteboard": "This job has been created using mrack."})
+
+        # RecipeSet attributes
+        specs.update({"priority": "Normal"})
+
+        # Add allowed keys
+        keypair = specs.get("keypair", None)
+        if keypair:
+            specs.update({"ks_append": self._allow_ssh_key(keypair)})
+
+        # Use ks_meta
+        specs.update({"ks_meta": "harness='restraint-rhts beakerlib-redhat'"})
+
+        # Recipe task definition
+        specs.update(
+            {  # we use dummy task because beaker reuire a task in recipe
+                "tasks": [{"name": "/distribution/dummy", "role": "STANDALONE"}]
+            }
+        )
+
+        # Create recipe with the specifications
+        recipe = BeakerRecipe(**specs)
+        recipe.addBaseRequires(**specs)
+
+        # Specify the architecture
+        arch_node = xml_doc().createElement("distro_arch")
+        arch_node.setAttribute("op", "=")
+        arch_node.setAttribute("value", specs["arch"])
+        recipe.addDistroRequires(arch_node)
+
+        # Add ReserveSys element to reserve system after provisioning
+        recipe.addReservesys(duration="86400")
```

Comment:
In the first version seems OK to have it hardcoded but later we will probably need to have reservation duration configurable.

Reply:
Yes that's correct

Reply:
Fixed